src/answers-umd.js

```diff
@@ -259,50 +259,29 @@ class Answers {
       if (parsedConfig.templateBundle) {
         this.renderer.init(parsedConfig.templateBundle);
       }
-
-      this._handlePonyfillCssVariables(
-        parsedConfig.disableCssVariablesPonyfill,
-        this._invokeOnReady.bind(this));
-      return this;
+    } else {
+      // Templates are currently downloaded separately from the CORE and UI bundle.
+      // Future enhancement is to ship the components with templates in a separate bundle.
+      this.templates = new DefaultTemplatesLoader(templates => {
+        this.renderer.init(templates);
+      });
     }
 
-    // Templates are currently downloaded separately from the CORE and UI bundle.
-    // Future enhancement is to ship the components with templates in a separate bundle.
-    this.templates = new DefaultTemplatesLoader(templates => {
-      this.renderer.init(templates);
-      this._handlePonyfillCssVariables(
-        parsedConfig.disableCssVariablesPonyfill,
-        this._invokeOnReady.bind(this));
+    const onReadyPromise = new Promise((resolve, reject) => {
```

Comment:
oh hmm sorry I missed this during the first round but could we just return the Promise returned by _handlePonyfillCssVariables? instead of wrapping it inside a new promise

Reply:
I wrapped it in a promise because the init resolving depends on the results of _handlePonyfillCssVariables promise and the MasterSwitchApi.isDisabled(), which is also a promise. If you can simplify it, I'm happy to take a look

Reply:
Update: We managed to simplify this!